extensions/ql-vscode/src/run-remote-query.ts

```diff
@@ -107,18 +104,44 @@ export async function runRemoteQuery(cliServer: cli.CodeQLCliServer, credentials
     return; // No error message needed, since `getRepositories` already displays one.
   }
 
-  await runRemoteQueriesApiRequest(credentials, ref, language, repositories, query);
+  // Get the controller repo from the config, if it exists.
+  // If it doesn't exist, prompt the user to enter it, and save that value to the config.
+  let controllerRepo: string | undefined;
+  controllerRepo = getRemoteControllerRepo();
+  if (!controllerRepo || !REPO_REGEX.test(controllerRepo)) {
+    void logger.log('No controller repository defined.');
```

Comment:
Should probably have a different error message if `REPO_REGEX.test` fails. Similar message to what you have below.

Maybe something like this:

```suggestion
    void logger.log(controllerRepo ? 'Invalid repository name.' : 'No controller repository defined.');
```

Reply:
Good point, thanks! 😎